src/aria/utils/Array.js

```diff
@@ -173,6 +173,30 @@ var Aria = require("../Aria");
                     }
                 }
                 return output;
+            },
+            /**
+             * This tests whether all elements in the array pass the test implemented by the provided function. 
+             * As long as the elements return truthy values, the loop will continue. However, as soon as a falsy
+             * value is returns, the answer will be returned without analyzed the rest of the elements.
+             * @param {Array} array Array to filter.
+             * @param {Function} Function to test each element of the array. This function receive as arguments the
+             * value, the index and the array.
+             * @param {Object} thisObject Object to use as this when executing callback.
+             */
+            every : (!arrayPrototype.every) ? function(array, callback, thisObject) {
+                // clone array to avoid mutation when executing the callback
+                var workArray = arrayUtils.clone(array);
+
+                thisObject = thisObject || array;
```

Comment:
Be careful that with this common shortcut for setting default values, you are preventing any falsy values to be used, even though they might be used in practice: `false`, `0`, `''` are all valid values for `thisArg`. Also, if the user wants to receive the global object as `thisArg`, he can't this way since he would have to specify `undefined` or `null`, which are both falsy values too.

Since the specifications http://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.every and the recommended polyfill implementation in MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every#Polyfill don't even match on that point, I don't know what to decide about this.

My general opinion is to leave it as flexible as possible, and let the code fail by itself later on (while calling `callback.call` in this specific case) if the value was not correct, in order to respect the DRY concept and to rely as most as possible on standard implementations.